### ISSUE TITLE
Conda pip interop fails with ~= dependencies

### ISSUE THREAD

**Issue by alanhdu** [conda/conda#7848]
## Current Behavior
Using `conda 4.6.0b0`, running `conda env export` sometimes returns:

```
MalformedVersionError: Malformed version string '~': invalid character(s)
```

### Steps to Reproduce
I am using `conda 4.6.0b0` with Python 3.7 on Ubuntu 18.04.

To reproduce this:
1. `conda create -n test python=3.7`
2. `conda activate test`
3. `pip install ipywidgets==7.4.2`
4. `conda env export`

will now fail.

By tracing through with pdb, I think the error is that `ipywidgets` has an `~=` dependency on `widgetsnbextension` (see https://github.com/jupyter-widgets/ipywidgets/blob/master/setup.py#L122) which `conda` is failing to parse and interpret correctly.

## Expected Behavior
Hopefully, `conda env export` should return without problems.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : test
    active env location : /opt/anaconda/envs/test
            shell level : 2
       user config file : /home/alan/.condarc
 populated config files : /home/alan/.condarc
          conda version : 4.6.0b0
    conda-build version : not installed
         python version : 3.7.0.final.0
       base environment : /opt/anaconda  (writable)
           channel URLs : https://conda.anaconda.org/conda-canary/linux-64
                          https://conda.anaconda.org/conda-canary/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /opt/anaconda/pkgs
                          /home/alan/.conda/pkgs
       envs directories : /opt/anaconda/envs
                          /home/alan/.conda/envs
               platform : linux-64
             user-agent : conda/4.6.0b0 requests/2.19.1 CPython/3.7.0 Linux/4.15.0-36-generic ubuntu/18.04.1 glibc/2.27
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
==> /home/alan/.condarc <==
channels:
  - conda-canary
  - conda-forge
  - defaults

```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at /opt/anaconda/envs/test:
#
# Name                    Version                   Build  Channel
backcall                  0.1.0                    pypi_0    pypi
bleach                    3.0.2                    pypi_0    pypi
bzip2                     1.0.6                h470a237_2    conda-forge
ca-certificates           2018.8.24            ha4d7672_0    conda-forge
certifi                   2018.8.24             py37_1001    conda-forge
decorator                 4.3.0                    pypi_0    pypi
defusedxml                0.5.0                    pypi_0    pypi
entrypoints               0.2.3                    pypi_0    pypi
ipykernel                 5.1.0                    pypi_0    pypi
ipython                   7.0.1                    pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
ipywidgets                7.4.2                    pypi_0    pypi
jedi                      0.13.1                   pypi_0    pypi
jinja2                    2.10                     pypi_0    pypi
jsonschema                2.6.0                    pypi_0    pypi
jupyter-client            5.2.3                    pypi_0    pypi
jupyter-core              4.4.0                    pypi_0    pypi
libffi                    3.2.1                hfc679d8_5    conda-forge
libgcc-ng                 7.2.0                hdf63c60_3    conda-forge
libstdcxx-ng              7.2.0                hdf63c60_3    conda-forge
markupsafe                1.0                      pypi_0    pypi
mistune                   0.8.4                    pypi_0    pypi
nbconvert                 5.4.0                    pypi_0    pypi
nbformat                  4.4.0                    pypi_0    pypi
ncurses                   6.1                  hfc679d8_1    conda-forge
notebook                  5.7.0                    pypi_0    pypi
openssl                   1.0.2p               h470a237_0    conda-forge
pandocfilters             1.4.2                    pypi_0    pypi
parso                     0.3.1                    pypi_0    pypi
pexpect                   4.6.0                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pip                       18.1                  py37_1000    conda-forge
prometheus-client         0.4.1                    pypi_0    pypi
prompt-toolkit            2.0.5                    pypi_0    pypi
ptyprocess                0.6.0                    pypi_0    pypi
pygments                  2.2.0                    pypi_0    pypi
python                    3.7.0                h5001a0f_4    conda-forge
python-dateutil           2.7.3                    pypi_0    pypi
pyzmq                     17.1.2                   pypi_0    pypi
readline                  7.0                  haf1bffa_1    conda-forge
send2trash                1.5.0                    pypi_0    pypi
setuptools                40.4.3                   py37_0    conda-forge
simplegeneric             0.8.1                    pypi_0    pypi
six                       1.11.0                   pypi_0    pypi
sqlite                    3.25.2               hb1c47c0_0    conda-forge
terminado                 0.8.1                    pypi_0    pypi
testpath                  0.4.2                    pypi_0    pypi
tk                        8.6.8                ha92aebf_0    conda-forge
tornado                   5.1.1                    pypi_0    pypi
traitlets                 4.3.2                    pypi_0    pypi
wcwidth                   0.1.7                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
wheel                     0.32.1                   py37_0    conda-forge
widgetsnbextension        3.4.2                    pypi_0    pypi
xz                        5.2.4                h470a237_1    conda-forge
zlib                      1.2.11               h470a237_3    conda-forge
```
</p></details>

**Comment by dhirschfeld** [comment]
Duplicate of #7776 and fixed by #7780.

@kalefranz - this can probably be closed.

**Comment by alanhdu** [comment]
Yup! I think this is fixed in `4.6.0b1`!
